x/storage/keeper/grpc_query.go

```diff
@@ -20,3 +20,19 @@ func (k Keeper) Params(c context.Context, req *types.QueryParamsRequest) (*types
 
 	return &types.QueryParamsResponse{Params: k.GetParams(ctx)}, nil
 }
+
+func (k Keeper) QueryParamsByTimestamp(c context.Context, req *types.QueryParamsByTimestampRequest) (*types.QueryParamsByTimestampResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+
+	params := k.GetParams(ctx)
+	versionedParams, err := k.GetVersionedParamsWithTs(ctx, req.GetTimestamp())
```

Comment:
If the timestamp is zero or not-specify, do we need to return the latest params?